web/users.go

```diff
@@ -1,45 +1,9 @@
 package web
 
 import (
-	"net/http"
-
 	pb "github.com/autograde/quickfeed/ag"
-	"github.com/autograde/quickfeed/database"
-	"github.com/jinzhu/gorm"
-	"github.com/labstack/echo/v4"
 )
 
-// JSONuser is a model to improve marshalling of user structure for authentication
-type JSONuser struct {
-	ID        uint64 `json:"id"`
-	IsAdmin   *bool  `json:"isadmin"`
-	Name      string `json:"name"`
-	StudentID string `json:"studentid"`
-	Email     string `json:"email"`
-	AvatarURL string `json:"avatarurl"`
-}
-
-// GetSelf redirects to GetUser with the current user's id.
-func GetSelf(db database.Database) echo.HandlerFunc {
```

Comment:
There is a `TestGetSelf` in `web/users_test.go` that now fails to compile. Can you rewrite `TestGetSelf` to work with the gRPC approach? You would probably need to call the `UserVerifier` interceptor or something, putting a dummy cookie/token into the context metadata, and checking that you get logged in. And similarly, that if you don't provide the required cookie, you won't be logged in.